runtime/syntax/objc.yaml

```diff
@@ -2,6 +2,7 @@ filetype: objective-c
 
 detect:
     filename: "\\.(m|mm|h)$"
+    signature: "(obj|objective)-c|#import|@(encode|end|interface|implementation|class|selector|protocol|synchronized|try|catch|finally|property|optional|required|import|autoreleasepool)"
```

Comment:
1. Shouldn't this signature be added to `objc.hdr` as well?
2. AFAICS this is the only actual functional change among the changes in the syntax files, right? Maybe do this change in a separate commit (again, just for the sake of ease of review)?
3. The ambiguity between C and C++ for header files is still not addressed, right? I.e. C still always takes precedence since it is alphabetically before C++?
4. This signature ensures (well, not really ensures but increases chances, but hey) that an Objective-C header file will be recognized as one. But how do we ensure that a C/C++ header is *not* recognized as an Objective-C one? Looks like we still just rely on the alphabetical order for that.

Reply:
1. The `*.hdr` files will be created at build time.
2. Yes, the first intention of this PR was to do a better guess which file type is present in case more than one FT matches. I try to separate as much as I can.
3. Yes, but now we've the capability to do so by adding identification marks.
4. Currently I'm unsure if `Regexp` was able to do the negative lookahead, but this would be something to prevent matches with marks of a different type.